app/src/main/java/com/clint/nasa/core/functional/Request.kt

```diff
@@ -0,0 +1,25 @@
+package com.clint.nasa.core.functional
+
+import com.clint.nasa.core.exception.Failure
+import retrofit2.Call
+
+object Request {
```

Comment:
Please keep an eye out for any race conditions if the same function is used multiple places.

Reply:
Sure.